packages/admin-ui/src/Select/useSelect.ts

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useMemo, useState } from "react";
+import { autorun } from "mobx";
+import { SelectProps } from "./Select";
+import { SelectPresenter } from "./SelectPresenter";
+
+export const useSelect = (props: SelectProps) => {
+    const presenter = useMemo(() => new SelectPresenter(), []);
+
+    const [vm, setVm] = useState(presenter.vm);
+
+    useEffect(() => {
+        presenter.init(props);
+    }, [props]);
```

Comment:
1. Initialize the presenter within the `useMemo` by passing only the required parameters.
2. It would be helpful to create a wrapper for this. Let's explore it further before implementing a wrapper for this repetitive code.
3. Yes, you are right! 👍